lib/wallet/wallet.js

```diff
@@ -3098,6 +3205,83 @@ class Wallet extends EventEmitter {
     return mtx;
   }
 
+  /**
+   * Make a finazling MTX for all transferring names
+   * @private
+   * @param {MTX?} mtx
+   * @param {Number?} witnessSize
+   * @returns {MTX}
+   */
+
+  async makeFinalizeAll(mtx, witnessSize) {
+    // Only allowed in makeBatch
+    assert(mtx, 'Batch MTX required for makeFinalizeAll.');
+    assert(witnessSize, 'Witness size required for batch size estimation.');
+    const height = this.wdb.height + 1;
+    const network = this.network;
+    const names = await this.getNames();
+
+    let finalizes = 0;
+    for (const ns of names) {
+      // Easiest check is for transfer state, do that first
+      if (!ns.transfer)
+        continue;
+
+      const blocksLeft = (ns.transfer + network.names.transferLockup) - height;
+      if (blocksLeft > 0)
+        continue;
+
+      // Then check for expiration
+      if (ns.isExpired(height, network))
+        continue;
+
+      // Now do the db lookups to see if we own the name
+      const {hash, index} = ns.owner;
+      const coin = await this.getUnspentCoin(hash, index);
+      if (!coin)
+        continue;
+
+      const version = coin.covenant.getU8(2);
+      const addr = coin.covenant.get(3);
+      const address = Address.fromHash(addr, version);
+
+      let flags = 0;
+
+      if (ns.weak)
+        flags |= 1;
+
+      const output = new Output();
+      output.address = address;
+      output.value = coin.value;
+      output.covenant.type = types.FINALIZE;
+      output.covenant.pushHash(ns.nameHash);
+      output.covenant.pushU32(ns.height);
+      output.covenant.push(Buffer.from(ns.name, 'ascii'));
+      output.covenant.pushU8(flags);
+      output.covenant.pushU32(ns.claimed);
+      output.covenant.pushU32(ns.renewals);
+      output.covenant.pushHash(await this.wdb.getRenewalBlock());
+
+      mtx.addOutpoint(new Outpoint(coin.hash, coin.index));
+      mtx.outputs.push(output);
+
+      // Keep batches below policy size limit
+      if (this.isOversizedBatch(mtx, witnessSize)) {
+        mtx.inputs.pop();
+        mtx.outputs.pop();
+        break;
+      }
+
+      // TODO: Should this factor of 6 be user-configurable?
+      // Enforce consensus limit per block at a maxmium
+      finalizes++;
+      if (finalizes > consensus.MAX_BLOCK_RENEWALS / 6)
+        break;
+    }
```

Comment:
Unlike other `make*All` functions that cap at 100 (MAX_BLOCK_* / 6), this check happens after adding a finalize so batches are capped at 101. Will either have to pop here or move it to the start of loop.